api_keys/file.go

```diff
@@ -0,0 +1,84 @@
+package apikeys
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+)
+
+type APIKeysFromFile struct {
+	FileName    string
+	users       map[string]string
+	lastUpdated time.Time
+	mu          sync.Mutex
+}
+
+func (k *APIKeysFromFile) readCSVToMap() (map[string]string, error) {
```

Comment:
This receiver can be a static function.

```suggestion
func readCSVToMap(filename string) (map[string]string, error) {
```